src/app/pages/signup/signup.page.ts

```diff
@@ -28,11 +27,11 @@ export class SignupPage implements OnInit {
   constructor(private router: Router, private activatedRoute: ActivatedRoute, public formBuilder: FormBuilder, private authService: AuthService, private googleAuthService: GoogleAuthService, private linkedinAuthService: LinkedinAuthService) {
     this.registerForm = new FormGroup({
       "mobile": new FormControl('', Validators.compose([
-        Validators.pattern('[0-9]+[0-9]+[0-9]+[0-9]+[0-9]+[0-9]+[0-9]+[0-9]+[0-9]+[0-9]{1,10}$'),
+        Validators.pattern('^[0-9]+[0-9]+[0-9]+[0-9]+[0-9]+[0-9]+[0-9]+[0-9]+[0-9]+[0-9]{1,10}$'),
```

Comment:
Remember: `^` represents the start of the pattern to match while `$` represents the end. Not strictly necessary but it basically prevents any characters from being ignored and forces the whole string to be considered for a match.

Reply:
Also this validator isn't right. It is basically an infinite sequence of numbers.

Reply:
Right,. So what's the appropriate way to approach validating a mobile number when signing up?